adapters/http-vertx/src/main/java/org/eclipse/hono/adapter/http/vertx/VertxBasedHttpProtocolAdapter.java

```diff
@@ -57,10 +59,20 @@ protected final void addRoutes(final Router router) {
         }
         addTelemetryApiRoutes(router);
         addEventApiRoutes(router);
+        setupCorsHandler(router);
+
+        router.options().handler(request -> request.response().setStatusCode(HttpResponseStatus.OK.code()).end());
+    }
+
+    private void setupCorsHandler(final Router router) {
+        router.route()
+                .handler(CorsHandler.create("*").allowedMethod(HttpMethod.PUT)
```

Comment:
I think there should be an option to limit the wildcard by using some configuration option. Not everyone might want the wildcard to be hardcoded.

Reply:
Good point. I will re-cook the patch

Reply:
I have put the property under "hono.http.corsAllowedOrigin"